Active_Record/app/models/booking.rb

```diff
@@ -0,0 +1,21 @@
+class Booking < ActiveRecord::Base
+  belongs_to :customer
+  belongs_to :sup
+
+  validates :start_date, :end_date, presence: true
+  validate :end_date_after_start_date
+
+  def booked_date_range
+    { from: :start_date, to: :end_date }
+  end
+
+  private
+
+  def end_date_after_start_date
+    return if end_date.blank? || start_date.blank?
```

Comment:
it is not necessary. This case is covered by `validates :start_date, :end_date, presence: true`

Reply:
done